components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs

```diff
@@ -713,7 +713,7 @@ private class StandaloneIdentity : IIdentity {
             Assert.Equal(2, jobs.Single()
                 .WriterGroup.DataSetWriters.Single()
                 .DataSet.DataSetSource.PublishedVariables.PublishedData.Single()
-                .HeartbeatInterval.Value.TotalSeconds);
+                .HeartbeatInterval.Value.TotalMilliseconds);
```

Comment:
>TotalMilliseconds [](http://example.com/codeflow?start=41&length=17)

I believe this was correct. we're testing for 2 seconds here

Reply:
I was trying to fix the inconsistency that publishing and sampling intervals were using milliseconds while heartbeat uses seconds. Is this intentional ? The other inconsistency was that publishing and sampling intervals are using int to store the value and `TimeSpan` is just a wrapper for it. Heartbeat does opposite of that, which I want to fix. Any reasons not to do that ? I can revert it to store seconds, but otherwise I would like int to store the actual value.

Reply:
The heartbeat was inherited from 2.5.x as being expressed in seconds. To a certain extent it makes sense to have it different than the publishing/sampling intervals because here we're typically envision the users to have much bigger intervals than the sampling. We cannot simply change the semantic of HeartbeatInterval in this version because it will break lots of already existing configurations for both 2.5.x and 2.6+. The idea was to have the heartbeat persisted by default as timespan just to smoothly transition from the seconds int to the more comprehensive timespan - also to cover the situation when users wanted to use second subdivisions as well.

Reply:
Reverted changes to ` OpcNodeModel.HeartbeatInterval`.